packages/itwin/grouping-mapping-widget/src/widget/components/ExtractionMessageModal.tsx

```diff
@@ -21,6 +24,42 @@ export interface ExtractionMessageModalProps {
 
 export const ExtractionMessageModal = ({ isOpen, onClose, extractionMessageData, timestamp }: ExtractionMessageModalProps) => {
   const [formattedTimestamp, setFormattedTimestamp] = useState<string>("");
+  const groupingMappingApiConfig = useGroupingMappingApiConfig();
+  const mappingClient = useMappingClient();
+  const [mappingId, setMappingId] = useState<string>("");
+  const [groupId, setGroupId]  = useState<string>("");
+  const { mappings } = useMappingsOperations({...groupingMappingApiConfig, mappingClient});
+  useEffect(() => {
+    extractionMessageData.forEach((extractionMessage) => {
+      if(extractionMessage.message.includes("iModel")){
+        extractionMessage.message = extractionMessage.message.replace(/iModel [\w-]+/, "iModel");
```

Comment:
It's not a good idea usually to directly modify objects like this. You will have to recreate the object and set it again.